### ISSUE TITLE
Testing is slower than Training

### ISSUE THREAD

**Issue by Colanim** [xcfcode/seq2seq-summarizer#5]
During training, the code is running at ~1.85 it/s

During testing, the code is running at ~1.4 it/s

It seems not a big different, but the time for testing is quite long : ~2h20

---

**Any idea from where it comes from ?**

As far as I understand, testing should be *much* faster : with `torch.no_grad()` there is no loss backward, so why the code is slow ? I'm wondering ^^

**Comment by xcfcode** [comment]
Is it related to beam_search?
